src/hera/workflows/io/_io_mixins.py

```diff
@@ -71,10 +73,16 @@ def _get_parameters(cls, object_override: Optional[Self] = None) -> List[Paramet
                 if param.name is None:
                     param.name = field
                 if param.default is not None:
-                    warnings.warn(
-                        "Using the default field for Parameters in Annotations is deprecated since v5.16"
-                        "and will be removed in a future minor version, use a Python default value instead"
-                    )
+                    if not global_config.experimental_features[_SUPPRESS_PARAMETER_DEFAULT_ERROR_FLAG]:
+                        warnings.warn(
```

Comment:
Should this not be outside the if to match the existing behaviour? (Or perhaps on the else path?)

Reply:
Yes good catch, I intended to keep the warning no matter the flag value